src/test/java/teammates/test/cases/webapi/GetStudentProfileActionTest.java

```diff
@@ -28,80 +30,232 @@ protected String getRequestMethod() {
+        StudentProfileAttributes expectedProfile = typicalBundle.profiles.get("student1InCourse1");
+        String expectedName = typicalBundle.accounts.get("student1InCourse1").name;
+        loginAsStudent(student1InCourse1.googleId);
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.COURSE_ID, student1InCourse1.getCourse(),
+        };
+        testGetCorrectProfile(expectedProfile, expectedName, submissionParams);
     }
 
-    private void testActionSuccess(AccountAttributes student, StudentProfileAttributes studentProfileAttributes,
-                                   String caseDescription) {
-        loginAsStudent(student.googleId);
+    @Test
+    public void testExecute_withStudentEmailAndCourseId_shouldReturnProfileByStudentEmail() {
+        StudentAttributes student1InCourse1 = typicalBundle.students.get("student1InCourse1");
+        StudentAttributes student2InCourse1 = typicalBundle.students.get("student2InCourse1");
+        StudentProfileAttributes expectedProfile = typicalBundle.profiles.get("student1InCourse1");
+        String expectedName = typicalBundle.accounts.get("student1InCourse1").name;
+        loginAsStudent(student2InCourse1.googleId);
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.STUDENT_EMAIL, student1InCourse1.getEmail(),
+                Const.ParamsNames.COURSE_ID, student1InCourse1.getCourse(),
+        };
+        testGetCorrectProfile(expectedProfile, expectedName, submissionParams);
+    }
 
-        ______TS(caseDescription);
+    @Test
+    public void testExecute_withProfileNotYetCreated_shouldReturnEmptyProfile() {
+        AccountAttributes student2InCourse1 = typicalBundle.accounts.get("student2InCourse1");
+        String expectedName = student2InCourse1.name;
+        StudentProfileAttributes expectedProfile = StudentProfileAttributes.builder(student2InCourse1.googleId).build();
+        loginAsStudent(student2InCourse1.googleId);
+        testGetCorrectProfile(expectedProfile, expectedName);
+    }
 
+    @Test
+    public void testExecute_getProfileOfUnregisteredStudent_shouldReturnNotFoundMessage() {
+        StudentAttributes student1InCourse1 = typicalBundle.students.get("student1InCourse1");
+        // Prepare an unregistered teammate
+        StudentAttributes unregisteredStudentInCourse1 =
+                StudentAttributes.builder(student1InCourse1.getCourse(), "student1InUnregisteredCourse@gmail.tmt")
+                .withGoogleId("")
+                .withName("unregistered student in course 1")
+                .withComment("")
+                .withSectionName(student1InCourse1.getSection())
+                .withTeamName(student1InCourse1.getTeam())
+                .build();
+        typicalBundle.students.put("unregisteredStudentInCourse1", unregisteredStudentInCourse1);
+        removeAndRestoreTypicalDataBundle();
+        removeAndRestoreDataBundle(typicalBundle);
```

Comment:
You can use `logic.createStudent()`.

Reply:
Actually the `logic` does not have this function, and I think it's unnecessary to create one just for testing

Reply:
But the creation of student is really weird.

`StudentsLogic.inst().createStudent`?